src/CommandLineOptions.ts

```diff
@@ -0,0 +1,51 @@
+import { Command } from 'commander'
+
+import packageJson from '../package.json'
+
+import * as lsif from './lsif'
+
+/** Configuration options to index a multi-project workspace. */
+export interface MultiProjectOptions {
+  inferTsconfig: boolean
+  progressBar: boolean
+  yarnWorkspaces: boolean
+  cwd: string
+  output: string
+  indexedProjects: Set<string>
+}
+
+/** Configuration options to index a single TypeScript project. */
+export interface ProjectOptions extends MultiProjectOptions {
+  projectRoot: string
+  projectDisplayName: string
+  writeIndex: (index: lsif.lib.codeintel.lsiftyped.Index) => void
+}
+
+export function commanderCommand(
+  indexAction: (projects: string[], otpions: MultiProjectOptions) => void
+): Command {
```

Comment:
```suggestion
export function mainCommand(
  indexAction: (projects: string[], options: MultiProjectOptions) => void
): Command {
```

Maybe name the function to be something more generic/describing its use, rather than the implementation? The fact that we're using the `commander` library is somewhat incidental.

Reply:
Great point, it was a bad name, renamed to `mainCommand`.